File: PaMVPDemo/app/src/main/java/com/example/wen/pamvpdemo/login/LoginPresenterImpl.java
package com.example.wen.pamvpdemo.login;

import com.pa.core.base.BasePresenter;
import com.pa.core.bean.TopResponse;
import com.pa.network.http.ApiSubscriberCallBack;

import io.reactivex.Flowable;

/**
 * Created by wen on 2018/5/14.
 */
public class LoginPresenterImpl extends BasePresenter<LoginContract.LoginInteractor, LoginContract.LoginView>
        implements LoginContract.LoginPresenter {


    private LoginContract.LoginInteractor loginInteractor;
    private LoginContract.LoginView loginView;
    private Flowable<TopResponse<Object>> userLogin;

    public LoginPresenterImpl(LoginContract.LoginView loginView,
                              LoginContract.LoginInteractor loginInteractor) {
        super(loginInteractor, loginView);
        this.loginInteractor = loginInteractor;
        this.loginView = loginView;
    }

    @Override
    public void login() {
        view.showLoadingView();
        userLogin = loginInteractor.getUserLogin();
        ApiSubscriberCallBack<Object> apiSubscriberCallBack = new ApiSubscriberCallBack<Object>() {

            @Override
            public void onSuccess(Object object) {
                loginView.onSuccess(object);
                view.hideLoadingView();
            }

            @Override
            public void onFailure(Throwable t) {
                loginView.onFailure();
                view.hideLoadingView();
            }
        };
        subscribe(userLogin,apiSubscriberCallBack);
    }
}

File: PaMVPKotlinDemo/app/src/main/java/com/example/wen/pamvpdemo/login/LoginPresenterImpl.kt
package com.example.wen.pamvpdemo.login

import com.pa.core.base.BasePresenter
import com.pa.core.bean.TopResponse
import com.pa.network.http.ApiSubscriberCallBack

import io.reactivex.Flowable

/**
 * Created by wen on 2018/5/14.
 */
class LoginPresenterImpl(private val loginView: LoginContract.LoginView,
                         private val loginInteractor: LoginContract.LoginInteractor) : BasePresenter<LoginContract.LoginInteractor, LoginContract.LoginView>(loginInteractor, loginView), LoginContract.LoginPresenter {
    private var userLogin: Flowable<TopResponse<Any>>? = null

    override fun login() {
        view!!.showLoadingView()
        userLogin = loginInteractor.userLogin()
        val apiSubscriberCallBack = object : ApiSubscriberCallBack<Any>() {

            override fun onSuccess(any: Any) {
                loginView.onSuccess(any)
                view?.hideLoadingView()
            }

            override fun onFailure(t: Throwable) {
                loginView.onFailure()
                view?.hideLoadingView()
            }
        }
        subscribe(userLogin!!, apiSubscriberCallBack)
    }
}
